tree/ntuple/v7/src/RNTupleDescriptorFmt.cxx

```diff
@@ -113,8 +113,8 @@ void ROOT::Experimental::RNTupleDescriptor::PrintInfo(std::ostream &output) cons
    std::uint64_t nPages = 0;
    int compression = -1;
    for (const auto &column : fColumnDescriptors) {
-      auto element = Detail::RColumnElementBase::Generate(column.second.GetModel().GetType());
-      auto elementSize = element.GetSize();
+      auto elementSize = Detail::RColumnElementBase::Generate(
```

Comment:
I am confused ... maybe ...   Why do we need to make an allocation just to get the element size?

Reply:
That's not great indeed.  We could define a static array that stores the on-disk sizes of an element of a given [column type](https://github.com/root-project/root/blob/master/tree/ntuple/v7/inc/ROOT/RColumnModel.hxx#L35). Or is this too brittle?

Reply:
I guess you could also duplicate the switch statement that is already in `RColumnElementBase::Generate`